src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -42,64 +42,7 @@ internal partial class DiagnosticIncrementalAnalyzer
                         return existingData;
                     }
 
-                    // PERF: Check whether we want to analyze this project or not.
-                    var fullAnalysisEnabled = GlobalOptions.IsFullSolutionAnalysisEnabled(project.Language, out var compilerFullAnalysisEnabled, out var analyzersFullAnalysisEnabled);
-                    if (forceAnalyzerRun)
-                    {
-                        // We are forcing full solution analysis for all diagnostics.
-                        fullAnalysisEnabled = true;
-                        compilerFullAnalysisEnabled = true;
-                        analyzersFullAnalysisEnabled = true;
-                    }
-
-                    if (!fullAnalysisEnabled)
```

Comment:
since fullAnalysisEnabled was always true (due to forceAnalyzerRun being true), this block was never run.